doc/man3/SSL_CTX_set1_curves.pod

```diff
@@ -263,6 +257,67 @@ returns -1.
 SSL_get_negotiated_group() returns the NID of the negotiated group used for
 key exchange, or NID_undef if there was no negotiated group.
 
+=head1 EXAMPLES
+
+Assume the server I<list> is "P-521:P-256/P-384/X25519:ffdhe2048" and client
+I<list> is "P-521:*P-384" when connecting to such a server, meaning that the
+client supports C<P-521> but does not send a key share for this group to the
+server, and the client supports C<P-384> including key share for this group.
+With server preference a HRR will be triggered for C<P-521> despite the
+availability of a key share for P-384, which overlaps with a lower priority
+server-side tuple.
+
+Consider the same server and server preference but different client with I<list>
+"P-521:P-384:*ffdhe2048". When the client attempts connecting to the server,
+only a keyshare for ffdhe2048 will be sent. But because P-521 and P-384 are
+supported by the client, an HRR will be triggered for P-521 (preferred by the
+server).
+
+As a separate example, consider a server I<list> "A:B/C:D/E:F". Listed in order
+of highest preference to least, 3 group tuples are created: "A:B", "C:D", and
+"E:F". Here are some examples of a client I<list> where setting server/client
+preference will not change the outcome:
+
+- "A:D:*F": Both prefer "A", but the server did not receive a key share for "A",
+so an HRR is triggered for "A".
+
+- "B:*C": Both prefer "B" from the first group tuple "A:B", so an HRR is
+triggered for "B".
+
+- "C:*F": Both prefer "C" from the second group tuple "C:D", so an HRR is
+triggered for "C".
+
+- "C:*D": Even though both prefer "C" over "D", the server will accept
+the key share for "D". Within a tuple, existing keyshares trump preference
+order.
+
+- "*C:*D": The server accepts the "C" key share.
+
+- "F": The server will trigger an HRR for "F" because it did not receive a key
+share for "F".
+
+Here are some examples where setting server/client preference will change the
+result:
+
+- "*D:*C"
+  - Client preference: The server uses the key share for "D".
+  - Server preference: The server uses the key share for "C".
+
+- "*E:C:A"
+  - Client preference: The server uses the key share for "E".
```

Comment:
This seems wrong? Don't we get an HRR for A in this case (i.e. same as for server preference)?